src/applications/edu-benefits/0993/containers/SubmitController.jsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
+
+import SubmitButtons from './SubmitButtons';
+import PrivacyAgreement from 'us-forms-system/lib/js/components/PrivacyAgreement';
+import { isValidForm } from 'us-forms-system/lib/js/validation';
+import {
+  createPageListByChapter,
+  expandArrayPages,
+  getActivePages,
+} from 'us-forms-system/lib/js/helpers';
+import {
+  setPrivacyAgreement,
+  setSubmission,
+  submitForm
+} from 'us-forms-system/lib/js/actions';
+
+class SubmitController extends React.Component {
+
+  componentWillReceiveProps(nextProps) {
+    const nextStatus = nextProps.form.submission.status;
+    const previousStatus = this.props.form.submission.status;
+    if (nextStatus !== previousStatus && nextStatus === 'applicationSubmitted') {
+      const newRoute = `${nextProps.formConfig.urlPrefix}confirmation`;
+      this.props.router.push(newRoute);
+    }
+  }
+
+  goBack = () => {
+    const {
+      form,
+      pageList,
+      router
+    } = this.props;
+
+    const eligiblePageList = getActivePages(pageList, form.data);
+    const expandedPageList = expandArrayPages(eligiblePageList, this.props.form.data);
+
+    router.push(expandedPageList[expandedPageList.length - 2].path);
+  }
+
+  handleSubmit = () => {
+    const {
+      form,
+      formConfig,
+      pagesByChapter
+    } = this.props;
+
+    const {
+      isValid
+    } = isValidForm(form, pagesByChapter);
+
+    if (isValid) {
+      this.props.submitForm(formConfig, form);
+    } else {
+      this.props.setSubmission('hasAttemptedSubmit', true);
```

Comment:
This component is the same as the us-forms-system SubmitController except for the removal of an if statement for handling validation messaging/logging validation errors on the review page (where errors may be hidden within a collapsed accordion), which is not necessary for this single page form as it does not include a review page:
```
      // validation errors in this situation are not visible, so we’d
      // like to know if they’re common
      if (privacyAgreementAccepted) {
        recordEvent({
          event: `${trackingPrefix}-validation-failed`,
        });
        Raven.captureMessage('Validation issue not displayed', {
          extra: {
            errors,
            prefix: trackingPrefix
          }
        });
        this.props.setSubmission('status', 'validationError');
      }
```